src/dhcp_proxy/dhcp_service.rs

```diff
@@ -149,46 +167,61 @@ impl From<DhcpServiceError> for Status {
     }
 }
 
-pub async fn process_client_stream(mut client: DhcpV4Service) {
-    while let Some(lease) = client.client.next().await {
-        match lease {
-            Ok(lease) => {
-                log::info!(
-                    "got new lease for mac {}: {:?}",
-                    &client.network_config.container_mac_addr,
-                    &lease
-                );
-                // get previous lease and check if ip addr changed, if not we do not have to do anything
-                if let Some(old_lease) = &client.previous_lease {
-                    if old_lease.yiaddr != lease.yiaddr
-                        || old_lease.subnet_mask != lease.subnet_mask
-                        || old_lease.gateways != lease.gateways
-                    {
-                        // ips do not match, remove old ones and assign new ones.
-                        log::info!(
-                            "ip or gateway for mac {} changed, update address",
-                            &client.network_config.container_mac_addr
-                        );
-                        match update_lease_ip(
-                            &client.network_config.ns_path,
-                            &client.network_config.container_iface,
-                            old_lease,
-                            &lease,
-                        ) {
-                            Ok(_) => {}
-                            Err(err) => {
-                                log::error!("{err}");
-                                continue;
+pub async fn process_client_stream(service_arc: Arc<Mutex<DhcpV4Service>>) {
+    loop {
```

Comment:
could this not just stay as loop like this
```
while let Some(lease) = service_arc.lock().await.client.next().await {
...
}
```

The extra scopes to reduce lock holding is interesting but not actually doing anything since this function is the only one holding the lock anyway.